learning.py

```diff
@@ -0,0 +1,81 @@
+import nn_architecture
+import data_loader as dl
+
+import logging
+import numpy as np
+import copy
+
+import torch
+from torch import nn
+import torch.optim as optim
+from torch.optim import lr_scheduler
+from torch.utils.data import Dataset, DataLoader
+
+
+def train(model, x, y, matrix):
+    best_loss = np.inf
+    loss_history = []
+
+    # Iterate throughout the epochs
+    for epoch in range(num_epochs):
+        # Zero the gradients
+        optimizer.zero_grad()
+        x_gpu = x.to(device)
+        y_gpu = y.to(device)
```

Comment:
These operations actually not fast. 
Should we from the start adopt more mature architecture, which utilize GPU better,  and train in **batches**, not in single pairs of scans?

Reply:
to be fixed in the next update